src/ripple/net/impl/RPCCall.cpp

```diff
@@ -497,14 +497,26 @@ class RPCParser
 
     bool isValidJson2(Json::Value const& jv)
     {
+        if (jv.isArray())
+        {
+            if (jv.size() == 0)
+                return false;
+            for (Json::UInt i = 0; i < jv.size(); ++i)
+            {
+                if (!isValidJson2(jv[i]))
+                    return false;
+            }
+            return true;
+        }
         if (jv.isObject())
         {
             if (jv.isMember(jss::jsonrpc) && jv[jss::jsonrpc] == "2.0" &&
                 jv.isMember(jss::ripplerpc) && jv[jss::ripplerpc] == "2.0" &&
                 jv.isMember(jss::id) && jv.isMember(jss::method))
             {
                 if (jv.isMember(jss::params) &&
-                      !(jv[jss::params].isArray() || jv[jss::params].isNull()))
+                      !(jv[jss::params].isArray() || jv[jss::params].isObject() ||
+                        jv[jss::params].isNull()))
```

Comment:
The `isNull()` call is superfluous as type is compared against `nullValue` with `isArray()` and `isObject()`

Reply:
Thanks!  Will fix.